### ISSUE TITLE
feature request: c11 Anonymous structures and unions

### ISSUE THREAD

**Issue by didickman** [AbsInt/CompCert#9]
gcc allows this c11 feature as a C extension:

```
$ cat foo.c
struct T { int tag; union { float x; int n; }; };

int f(){
        struct T var;
        var.x = 3.0;
}
$ gcc -c foo.c
$ gcc -std=c99 -c foo.c
foo.c:1: warning: declaration does not declare anything
foo.c: In function 'f':
foo.c:5: error: 'struct T' has no member named 'x'
$ gcc -std=gnu99 -c foo.c
$
```

CompCert doesn't currently support this feature:

```
$ ccomp -fall -c foo.c
foo.c:5: Error: struct 'T' has no member named 'x'.
Fatal error; compilation aborted.
1 error detected.
```

Here's the gcc documentation:
  https://gcc.gnu.org/onlinedocs/gcc/Unnamed-Fields.html

**Comment by xavierleroy** [comment]
Still no support for this feature in CompCert.  However, commit [master 7dd10e8] adds a descriptive error message when an anonymous struct/union is used as a member of a struct/union.  Before, this was just ignored like another declaration that declares zero members.

**Comment by didickman** [comment]
Thanks. I do see the nicer error message now. That is helpful.

```
/usr/src/libexec/tradcpp/macro.c:42: Error:
  ISO C99 does not support anonymous structs/unions.
/usr/src/libexec/tradcpp/macro.c:80: Error:
  struct 'expansionitem' has no member named 'string'.
Fatal error; compilation aborted.
2 errors detected.
```

**Comment by bschommer** [comment]
Is implemented and merged with commit 9b022d749dc5b1fdb091ff8a74647c03fc1f84bb.
